asynq/decorators.py

```diff
@@ -146,16 +150,53 @@ def asynq(self, *args, **kwargs):
 class AsyncDecorator(PureAsyncDecorator):
     binder_cls = AsyncDecoratorBinder
 
+    def __init__(self, fn, cls, kwargs={}, asyncio_fn=None):
+        super().__init__(fn, cls, kwargs)
+        self.asyncio_fn = asyncio_fn
+
     def is_pure_async_fn(self):
         return False
 
     def asynq(self, *args, **kwargs):
+        if asynq_to_async.is_asyncio_mode():
```

Comment:
This may turn out to be expensive since we have to do this check on every `.asynq()` call. I think we should be able to avoid an "asyncio mode" if `resolve_awaitables` can simply deal with the AsyncTask object that `_call_pure` normally produces: it will have to pull out the underlying generator and wrap it.

If we do need this, it would be good to add a `.pxd` file for `asynq_to_async.py` and compile it with Cython. That will likely make this call very cheap, just a C call and integer comparison.

Reply:
I tried .pxd, a duplicated implementation was required for inlining, resulting in re-definition compile error. The solution required the modification of the `.py` file. I ended up with a single .py file with a few cython annotations.
```
if not cython.compiled:
   def is_asyncio_mode():
     ..
```